### PR TITLE
Add plutil utility

### PR DESCRIPTION
**Pull request by Artoria2e5**
This is supposedly an expansion of the macOS plutil(1). See #101.

- [x] Main utility
- [x] extra formats:
  - [ ] write `objc` (modern literal format)
  - [ ] write `swift`
  - [x] read/write NSJSONSerialization 
- [ ] tests (shell script)
- [ ] manpage
- [ ] gsautodoc

### CODE REVIEW

File: Source/NSPropertyList.m [hunk 1]
```diff
@@ -345,7 +345,7 @@ - (void) parser: (NSXMLParser *)parser
     }
   else if ([elementName isEqualToString: @"real"])
     {
-      ASSIGN(plist, [NSNumber numberWithDouble: [value doubleValue]]);
+      ASSIGN(plist, [NSNumber numberWithDouble: strtod([value cString], 0)]);
```
Comment (ngrewe) [inline Source/NSPropertyList.m]: Cool! A minor (teensy, in fact) nitpick, though: The type of the second parameter to `strtod` is also a pointer, so `NULL` would be more appropriate than `0`.

Comment (Artoria2e5) [reply Source/NSPropertyList.m]:
```suggestion
      ASSIGN(plist, [NSNumber numberWithDouble: strtod([value cString], NULL)]);
```

**Comment by rfm** [comment]
I note that build is failing because at plutil.m lines 107 and 109 there is a comparison between a pointer to a pointer and nil.  This comparison should use 0 rather than nil.

**Comment by probonopd** [comment]
Is this still being worked on? I am also running into the same compilation issue.

**Comment by Artoria2e5** [comment]
Yes! Sorry for all the delays…

**Comment by Artoria2e5** [comment]
Okay, it compiles now. I am going to leave the cool new language formats unimplemented, and work on the doc and the tests instead.

**Comment by probonopd** [comment]
Thank you very much. This is a very useful tool. I hope it gets added at some point.

**Comment by rfm** [comment]
Thanks very much ... merged.

**Pull request merged**